libpam/pam_google_authenticator.c

```diff
@@ -350,10 +352,22 @@ static int open_secret_file(pam_handle_t *pamh, const char *secret_filename,
     return -1;
   }
 
+  // print comparisson
+  if (params->debug) {
+    log_message(LOG_INFO, pamh, "Secret file permissions are %4o", orig_stat->st_mode & 03577);
+    if (params->allow_perm) {
+      log_message(LOG_INFO, pamh, "Extra file permissions allowed are %4o", params->allow_perm);
+    }
+  }
+
   // Check permissions on "~/.google_authenticator"
-  if ((orig_stat->st_mode & 03577) != 0400 ||
+  if (((orig_stat->st_mode & 03577) != 0400 &&
```

Comment:
Yes, that sounds better. It wouldn't allow un-setting owner-readable, but then again that wouldn't make any sense to me.

One slight fix though: Change 0400 into 03400, and keep the second logical and inside the parenthesis.